astroquery/desi/core.py

```diff
@@ -0,0 +1,97 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import urllib.error
+import requests
+import warnings
+
+import pyvo as vo
+import numpy as np
+import astropy.coordinates as coord
+
+from astroquery.exceptions import NoResultsWarning
+from astroquery.query import BaseQuery
+from astroquery.utils import commons, async_to_sync
+from astroquery.desi import conf
+
+__all__ = ['DESILegacySurvey', 'DESILegacySurveyClass']
+
+
+class DESILegacySurveyClass(BaseQuery):
+
+    def query_region(self, coordinates, radius, *, data_release=9):
+        """
+        Queries a region around the specified coordinates.
+
+        Parameters
+        ----------
+        coordinates : `astropy.coordinates`
+            coordinates around which to query.
+        radius : `astropy.units.Quantity`
+            the radius of the cone search.
+        data_release: int
+            the data release of the LegacySurvey to use.
+
+        Returns
+        -------
+        response : `astropy.table.Table`
+        """
+
+        tap_service = vo.dal.TAPService(conf.tap_service_url)
+        coordinates_transformed = coordinates.transform_to(coord.ICRS)
+
+        qstr = (f"SELECT all * FROM ls_dr{data_release}.tractor WHERE "
+               f"dec>{coordinates_transformed.dec.deg - radius.deg} and "
+               f"dec<{coordinates_transformed.dec.deg + radius.deg} and "
+               f"ra>{coordinates_transformed.ra.deg - radius.deg / np.cos(coordinates_transformed.dec.deg * np.pi / 180.)} and "
+               f"ra<{coordinates_transformed.ra.deg + radius.deg / np.cos(coordinates_transformed.dec.deg * np.pi / 180)}")
+
+        tap_result = tap_service.run_sync(qstr)
+        tap_result = tap_result.to_table()
+        # filter out duplicated lines from the table
+        mask = tap_result['type'] != 'D'
+        filtered_table = tap_result[mask]
+
+        return filtered_table
+
+    def get_images(self, position, pixels, radius, *, data_release=9, show_progress=True, image_band='g'):
+        """
+        Downloads the images for a certain region of interest.
+
+        Parameters
+        -------
+        position: `astropy.coordinates`.
```

Comment:
Do add all the arguments to the docstring. Also, pixel and radius can be optional.